napari/layers/_shapes_layer/model.py

```diff
@@ -441,9 +445,17 @@ def _get_shape(self):
         """Determines the shape of the vertex data.
         """
         if len(self.data._vertices) == 0:
-            return [1, 1]
+            slice_shape = [1, 1]
+        else:
+            slice_shape = list(np.max(self.data._vertices, axis=0) + 1)
+
+        if len(self.nd_data.keys()) == 1:
+            return [1] * (self.nd_data_dim-2) + slice_shape
         else:
-            return np.max(self.data._vertices, axis=0) + 1
+            keys = list(self.nd_data.keys())
+            keys.remove('2D')
```

Comment:
This feels super hacky. The dictionary should only have one type of thing in it.

Reply:
yes, replacing with empty tuple